crawl-ref/source/webserver/server.py

```diff
@@ -18,7 +18,10 @@
 class MainHandler(tornado.web.RequestHandler):
     def get(self):
         host = self.request.host
-        if self.request.protocol == "https":
+        proto = self.request.headers.get_list('x-forwarded-proto')
```

Comment:
Since you only care about the latest value of this header if it is present, you can simplify the code to:

```py
if self.request.protocol == "https" or self.request.headers.get("x-forwarded-proto", "http") == "https":
```

Reply:
I couldn't figure out how to do that so simple like headers.get()! The documentation described it as a "dictionary-like object" and get_list() was the only getter they told me about.